andela/travel_tool_front
# #159725717 Integrate the Requests Table to the Backend

#### What does this PR do?
Integrate the requests table with the backend in order to get data from the database

#### Description of Task to be completed?
- Create actions for fetching requests
- Create sagas for performing asynchronous actions
- Create services for making API call and handling error
- Create preloader component
- Create HOC for adding loading to components
- Create dropdown for selecting data limit
- Implement filtering of requests i.e all, open and past requests
- Write tests for reducers, actions, sagas, services and components

#### How should this be manually tested?
- Clone the repo -` git clone https://github.com/andela/travel_tool_front.git`
- Enter into the project directory - `cd travel_tool_front`
- Start the app using `make start`
- Start the API - see (https://github.com/andela/travel_tool_back/pull/18)
- To be able to login, add a supported domain name to your hosts:
`vim /etc/hosts` and add the line 127.0.0.1 local.andela.com
- Open the browser and navigate to the Requests URL `http://127.0.0.1:3000/requests`
- You should see your requests if there are any.

#### Any background context you want to provide?
This feature relates with (https://github.com/andela/travel_tool_front/pull/43) to add requests to the database.

#### What are the relevant pivotal tracker stories?
[#159725717](https://www.pivotaltracker.com/story/show/159725717)

#### Screenshots (if appropriate)
None

#### Questions:
N/A
